tornado-runtime/src/main/java/uk/ac/manchester/tornado/runtime/common/TornadoOptions.java

```diff
@@ -243,14 +243,29 @@ public class TornadoOptions {
      * default.
      */
     public static final boolean ENABLE_STREAM_OUT_BLOCKING = getBooleanValue("tornado.enable.streamOut.blocking", TRUE);
+
     /**
      * Option to run concurrently on multiple device in single or multi-backend
      * configuration. False by default.
      */
     public static final boolean CONCURRENT_INTERPRETERS = Boolean.parseBoolean(System.getProperty("tornado.concurrent.devices", "False"));
+
+    /**
+     * Panama Object Header in TornadoVM.
+     */
     public static final long PANAMA_OBJECT_HEADER_SIZE = TornadoNativeArray.ARRAY_HEADER;
+
+    /**
+     * Option to define the maximum number of internal events related to OpenCL/PTX/SPIR-V events to keep alive.
+     * This is application specific. In a large application, there are probably many events that need to keep alive
+     * to perform the sync.
+     */
+    public static final int MAX_EVENTS = getIntValue("tornado.max.events", "32768");
```

Comment:
Does this number have any upper bound, being vendor-specific?

Reply:
This number is application specific.

Reply:
so, there is not constraint from driver side regarding the max number of events that it can serve?

Reply:
No, TornadoVM has this constraint to try not to pre-allocate much memory for the events. So there is an upper limit and then but events-memory events are reused from a circular buffer. This PR just increases the limit to accommodate larger applications, which was the case of GAIA.